src/overlays/actors/ovl_En_Fr/z_en_fr.c

```diff
@@ -27,99 +183,1134 @@ const ActorInit En_Fr_InitVars = {
+
+        EnFr_DrawActive(this);
+
+        this->actionFunc = EnFr_JumpingOutOfWater;
+    }
+}
+
+void EnFr_JumpingOutOfWater(EnFr* this, GlobalContext* globalCtx) {
+    Vec3f vec1;
+    Vec3f vec2;
+
+    // Increment flag to time the next frog jumping out of water
+    if (this->skelAnime.animCurrentFrame == 6.0f) {
+        sEnFrPointers.flags++;
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+
+        // Midway through jump
+    } else if (this->skelAnime.animCurrentFrame == 3.0f) {
+        this->actor.gravity = -10.0f;
+        this->actor.speedXZ = 0.0f;
+        this->actor.velocity.y = 47.0f;
+    }
+
+    vec1.y = 0.0f;
+    vec1.x = 0.0f;
+    vec1.z = this->xzDistToLogSpot;
+
+    Matrix_RotateY(((this->actor.posRot.rot.y + 32768) / 32768.0f) * M_PI, MTXMODE_NEW);
+    Matrix_MultVec3f(&vec1, &vec2);
+
+    this->actor.posRot.pos.x = this->posLogSpot.x + vec2.x;
+    this->actor.posRot.pos.z = this->posLogSpot.z + vec2.z;
+
+    if (this->skelAnime.animCurrentFrame >= 3.0f) {
+        Math_SmoothScaleMaxF(&this->xzDistToLogSpot, 0.0f, 1.0f, 10.0f);
+    }
+
+    // The frog has reached the log
+    if (EnFr_IsBelowLogSpot(this, &vec2.y)) {
+        this->actor.gravity = 0.0f;
+        this->actionFunc = EnFr_OrientOnLogSpot;
+        this->unusedFloat = 0.0f;
+    }
+
+    if ((this->actor.velocity.y <= 0.0f) && (vec2.y < 40.0f)) {
+        this->skelAnime.animPlaybackSpeed = 1.0f;
+    }
+}
+
+void EnFr_OrientOnLogSpot(EnFr* this, GlobalContext* globalCtx) {
+    s16 func;
+    func = Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 2, 10000, 100);
+    this->actor.posRot.rot.y = this->actor.shape.rot.y;
+    if (func == 0 && this->skelAnime.animCurrentFrame == this->skelAnime.animFrameCount) {
+        sEnFrPointers.flags++;
+        this->actionFunc = EnFr_ChooseJumpFromLogSpot;
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06001534, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06001534), 0, 0.0f);
+    }
+}
+
+void EnFr_ChooseJumpFromLogSpot(EnFr* this, GlobalContext* globalCtx) {
+    // Jumping back into water
+    if (sEnFrPointers.flags == 0xC) {
+        this->actor.posRot.rot.y = (32768.0f / M_PI) * sLogSpotToFromWater[this->actor.params].yaw;
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060007BC, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060007BC), 2, 0.0f);
+
+        this->actionFunc = EnFr_JumpingBackIntoWater;
+    } else if (this->isJumpingUp) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_060007BC, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_060007BC), 2, 0.0f);
+        this->actionFunc = EnFr_JumpingUp;
+    }
+}
+
+void EnFr_JumpingUp(EnFr* this, GlobalContext* globalCtx) {
+    f32 yDistToLogSpot;
+
+    if (this->skelAnime.animCurrentFrame == 6.0f) {
+        this->skelAnime.animPlaybackSpeed = 0.0f;
+
+    } else if (this->skelAnime.animCurrentFrame == 3.0f) {
+        this->actor.gravity = -10.0f;
+        this->actor.velocity.y = 25.0f;
+
```

Comment:
in general in this file id remove the line breaks between different blocks in the same if. i personally like doing so between different if blocks, but within the same one it makes it hard to follow i feel